### ISSUE TITLE
Test Agnus/VPOS/probe4 fails

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#640]
This one is interesting: 

A500 8A ECS: 🥰

![probe4_A500_ECS](https://user-images.githubusercontent.com/12561945/151550546-0f9304a3-1b36-497e-ab5d-29f9738d3bf0.JPG)

The test reads VHPOSR 16 times in a scanline passing the right border with the BPLCON0::ERSY bit set to 1. ERSY enabled external synchronisation which means the Amiga stops generating HSYNC and VSYNC signals as long as the bit is set (hence the display corruption). The test has been written to prove the currently implemented ERSY latching mechanism in vAmiga to be wrong (which it does):

<img width="976" alt="Bildschirmfoto 2022-01-28 um 13 58 29" src="https://user-images.githubusercontent.com/12561945/151551061-8a368a16-ca38-41c0-bae5-435aac47e9ba.png">

What I am puzzled about is the first blue stripe in the original image. The stripes visualise the loop counter used to write the measured result values into the bitmap. The large size of the first stripe indicates that the CPU is slowed down massively on the real machine in this area. I have no idea where this comes from.

**Comment by dirkwhoffmann** [comment]
Values are fixed. Still unsure about the large blue stripe...

<img width="976" alt="Bildschirmfoto 2022-01-28 um 14 15 36" src="https://user-images.githubusercontent.com/12561945/151553196-733e8a4a-d088-473f-a4b0-600221e04153.png">

**Comment by mras0** [comment]
How are you creating the blue gradient? Maybe I'm not following the source code correctly (https://github.com/dirkwhoffmann/vAmigaTS/blob/master/Agnus/VPOS/probe4/probe4.s) but I don't see where the relevant portion is? (Small aside: it looks like textutil.i isn't in the repo).

FWIW this is what it looks like on a a1200 (I know you're not doing AGA/020+ stuff, but maybe it can help in figuring out where the time is/isn't spent):

With 060/50MHz accelerator:
![IMG_20220128_175937](https://user-images.githubusercontent.com/9952017/151593400-3c8901ca-b853-48bc-9a37-f230c1e031bd.jpg)

Plain A1200:
![IMG_20220128_180011](https://user-images.githubusercontent.com/9952017/151593408-24ab1f89-dbb4-4f86-a50a-bb805bc7b106.jpg)

**Comment by dirkwhoffmann** [comment]
```
   moveq   #15,d2              ; Line counter
.l:
    ; Read measured value
    move.w  d2,$DFF180
    moveq   #0,d0
    move.w  (a2)+,d0

    ; Compare with the expected value and select the target bitplane accordingly
    lea     bitplane1+2,a0
    cmp.w   (a5)+,d0
    bne     .skip
    lea     bitplane2+2,a0
.skip:
    ; Print line
```

The second move writes the line counter into the color register. I did this to verify that the code finishes within a single frame. 

I think the "large color bar issue" is something really simple. The first color bar only appears to be larger, because of the color range of my monitor. It's multiple bars that appear as a single one (I still need to verify this by using different colors though). 

It's interesting to see that it's different on the A1200. Red color means that the values don't match the expected ones. 

You might be curious why I wrote this test: My old implementation latched the position counter whenever the ERSY bit is set to 1. I thought I can use this feature to measure the exact position of the Copper (in order to see if certain DMA cycles get skipped or not). My idea was to wait for a specific cycle and set the ERSY bit with the Copper. After that, I would trigger an interrupt where I read the latched position counter and print it out. Unfortunately, this doesn't work because no real latching takes place on the real machine. 

> Small aside: it looks like textutil.i isn't in the repo

Fixed. BTW, `textUtil.i` is inspired by your test program for printing out the initial values of the CPU registers.

**Comment by mras0** [comment]
> The second move writes the line counter into the color register. I did this to verify that the code finishes within a single frame.

Ah right, missed that on reading since due to the direct register write (was just skimming and searching for COLOR00).
 
> I think the "large color bar issue" is something really simple. The first color bar only appears to be larger, because of the color range of my monitor. It's multiple bars that appear as a single one (I still need to verify this by using different colors though).

Probably, but wouldn't you have noticed that on other tests? The color fidelity of OCS/ECS is great. I can't test on my A600 ATM though, as it's in too bad shape (battery leak :( ) and I only have one gotek drive that I don't dare to move right now.

> You might be curious why I wrote this test: My old implementation latched the position counter whenever the ERSY bit is set to 1. I thought I can use this feature to measure the exact position of the Copper (in order to see if certain DMA cycles get skipped or not). My idea was to wait for a specific cycle and set the ERSY bit with the Copper. After that, I would trigger an interrupt where I read the latched position counter and print it out. Unfortunately, this doesn't work because no real latching takes place on the real machine.

Looks like a good test case, and FWIW WinUAE 4.9.1b5 also fails it.

**Comment by dirkwhoffmann** [comment]
Confirmed. It's a "fake issue". With a modified test that shifts the line counter two bits to the left before writing it into the color register, all bars are distinguishable: 

![IMG_2336](https://user-images.githubusercontent.com/12561945/151605163-7f5c4fd0-1c82-45af-a59b-bc5905d46ef5.JPG)

<img width="976" alt="Bildschirmfoto 2022-01-28 um 19 42 30" src="https://user-images.githubusercontent.com/12561945/151605197-3eeaff1d-1811-4bbf-b02d-b6cbfee20913.png">

The image on the TFT is much brighter than the image in vAmiga.

**Comment by dirkwhoffmann** [comment]
Fixed in v2.0b1 which is now online.
